src/services/peerconnection.ts

```diff
@@ -174,12 +171,68 @@ export class PeerConnectionHelper {
     }
 
     /**
-     * Create a new secure data channel.
+     * Initiate the handover process.
      */
-    public createSecureDataChannel(label: string): saltyrtc.tasks.webrtc.SecureDataChannel {
-        const dc: RTCDataChannel = this.pc.createDataChannel(label);
+    public handover(): void {
+        if (this.signalingChannel !== null) {
+            throw new Error('Handover already inintiated');
+        }
+
+        // Get transport link
+        const link: saltyrtc.tasks.webrtc.SignalingTransportLink = this.webrtcTask.getTransportLink();
+
+        // Create data channel
+        const dc = this.pc.createDataChannel(link.label, {
+            id: link.id,
+            negotiated: true,
+            ordered: true,
+            protocol: link.protocol,
+        });
         dc.binaryType = 'arraybuffer';
-        return this.webrtcTask.wrapDataChannel(dc);
+
+        // Wrap as an unbounded, flow-controlled data channel
+        this.signalingChannel = new UnboundedFlowControlledDataChannel(this.$log, dc);
+
+        // Create transport handler
+        const self = this;
+        const handler = {
+            get maxMessageSize(): number {
+                return self.pc.sctp.maxMessageSize;
+            },
+            close(): void {
+                self.$log.debug(self.logTag, `Signalling data channel close request`);
+                dc.close();
+            },
+            send(message: Uint8Array): void {
+                self.$log.debug(self.logTag, `Signalling data channel outgoing signaling message of ` +
+                    `length ${message.byteLength}`);
+                self.signalingChannel.write(message);
+            },
+        };
+
+        // Bind events
+        dc.onopen = () => {
+            this.$log.info(this.logTag, `Signalling data channel open`);
+
+            // Rebind close event
+            dc.onclose = () => {
+                this.$log.info(this.logTag, `Signalling data channel closed`);
+                link.closed();
+            };
```

Comment:
Just so I understand this: Why do we rebind onclose but not onerror or onmessage?

Reply:
The semantic of `close` changes (*should not happen* -> *valid*) after the channel opened but the semantic of `error` remains the same (*should not happen*).

Reply:
Ah, of course :slightly_smiling_face: